Library/Homebrew/version.rb

```diff
@@ -179,6 +179,19 @@ def self.detect(url, specs)
     end
   end
 
+  # TODO add tests if method is going to be left
+  def self.parse_version(val)
```

Comment:
I agree that using `parse` for that would be complete overkill. Maybe a better approach than creating this method would be to add a `Version.create` that dispatches to either `Version.new` or `HeadVersion.new` depending on the given string. This would also avoid some duplication of this logic in other code from this PR.